test/jdk/java/awt/ScrollPane/ScrollPaneInsets/ScrollPaneExtraScrollBar.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2003, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+  @test
+  @bug 4152524
+  @summary Test that scroll pane doesn't have scroll bars visible when it is shown for the first time with SCROLLBARS_AS_NEEDED style
+  @key headful
+*/
+
+import java.awt.Button;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Insets;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.ScrollPane;
+
+import java.awt.event.InputEvent;
+
+public class ScrollPaneExtraScrollBar
+{
+    ScrollPane sp;
+    Frame f;
+    Robot robot;
+
+    public static void main(String[] args) throws Exception {
+        ScrollPaneExtraScrollBar scrollTest = new ScrollPaneExtraScrollBar();
+        scrollTest.init();
+        scrollTest.start();
+    }
+    public void init() throws Exception {
+
+       robot = new Robot();
+
+        EventQueue.invokeAndWait(() -> {
+                f = new Frame();
+                sp = new ScrollPane(ScrollPane.SCROLLBARS_AS_NEEDED);
+                sp.add(new Button("TEST"));
+                f.add("Center", sp);
+                f.pack();
+                f.setLocationRelativeTo(null);
+                f.setVisible(true);
+        });
+    }
+
+    public void start() throws Exception {
+        try {
+            if (System.getProperty("os.name").startsWith("Windows")) {
```

Comment:
Should we use [`OSInfo` class](https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/sun/awt/OSInfo.java)?

It's used in [`bug4847375.java`](https://github.com/openjdk/jdk/blob/master/test/jdk/javax/swing/JFileChooser/4847375 /bug4847375.java), [`bug6550546.java`](https://github.com/openjdk/jdk/blob/master/test/jdk/javax/swing/JFileChooser/6550546/bug6550546.java) and others.

The `@requires` tag should be added to the test tags so that it's not selected for other platforms.

To avoid extensive indentation, reverse the condition in the `if` statement and return early.

Reply:
Updated.